### PR TITLE
[automate-2514] add v2 prefix to IAM API paths

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :nut_and_bolt: Description: What code changed, and why?
In preparation for the GA Release of IAM v2, we're updating 

### :+1: Definition of Done
- [x] all IAM v2 APIs now have a primary HTTP path with the prefix `iam/v2`
- [x] the old IAM `iam/v2beta` paths still work (those will be removed on our force-upgrade a month after the GA Release)
- [x] update the IAM paths in the UI to `iam/v2` (otherwise this change would break v2 introspection)
- [ ] follow-up PRs will update the integration tests and studio helpers to replace `v2beta` paths with `v2`

### :athletic_shoe: How to Build and Test the Change
```
rebuild components/automate-gateway
rebuild components/automate-cli

# the grpc endpoints are now under v2 instead of v2beta
chef-automate dev grpcurl automate-gateway chef.automate.api.iam.v2.Policies.ListRoles 

echo '{"id": "editor" }' | chef-automate dev grpcurl automate-gateway -- -d @ chef.automate.api.iam.v2.Policies.GetRole 

export TOK=`chef-automate iam token create ADM --admin`

# make a policy and post it
echo '{
   "name": "testing",
   "id": "test1",
   "members": [
     "user:local:coolbean"
   ],
   "statements": [
     {
       "effect": "DENY",
       "actions": ["iam:*:list"],
       "projects": [
         "*"
       ]
     }
   ],
   "projects": []
 }' >> pol.json

curl -kH "api-token: $TOK" -X POST -d @pol.json https://localhost/apis/iam/v2beta/policies | jq
# success
curl -kH "api-token: $TOK" -X POST -d @pol.json https://localhost/apis/iam/v2/policies | jq
# this results in a conflict error unless you change the id in pol.json

curl -kH "api-token: $TOK"  https://localhost/apis/iam/v2beta/policies/test1 | jq
curl -kH "api-token: $TOK"  https://localhost/apis/iam/v2/policies/test1 | jq

curl -kH "api-token: $TOK" -X PUT -d @pol.json https://localhost/apis/iam/v2beta/policies/test1 | jq
curl -kH "api-token: $TOK" -X PUT -d @pol.json https://localhost/apis/iam/v2/policies/test1 | jq

curl -kH "api-token: $TOK" -X DELETE https://localhost/apis/iam/v2beta/policies/test1 | jq
curl -kH "api-token: $TOK" -X DELETE https://localhost/apis/iam/v2/policies/test1 | jq
```
Play around with any of the APIs, they should all work with both paths.

Click around the Settings UI, make sure the sidebar renders all the menu items.

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [x] Docs added/updated?

### CODE REVIEW

**Comment by lancewf** [comment]
Why did the automate-ui code have to be updated in the PR?

File: components/automate-gateway/api/iam/v2/policy.proto [hunk 1]
```diff
@@ -15,149 +15,258 @@ import "components/automate-grpc/protoc-gen-policy/iam/annotations.proto";
 
 service Policies {
   rpc CreatePolicy (CreatePolicyReq) returns (CreatePolicyResp) {
-    option (google.api.http).post = "/iam/v2beta/policies";
-    option (google.api.http).body = "*";
+    option (google.api.http) = {
```
Comment (msorens) [inline components/automate-gateway/api/iam/v2/policy.proto]: Tried to look up the syntax for `option` in the language specification (https://developers.google.com/protocol-buffers/docs/reference/proto3-spec) but... it is missing!

However, a few paragraphs into https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#google.api.HttpRule, there are at least precedents for us to follow. Based on that, commas should not be used within `option` objects:

Mostly, you have correctly omitted commas, but there are some lurking that should be deleted:
L22
L35
L59
L72
L110
L124
L138
L152
L202
L216
L230

(Similar corrections needed in other proto files.)

**Comment by bcmdarroch** [comment]
@lancewf I ended up updating the UI code here because I had to update the introspection paths (because the paths we pass to introspection in the UI need to match the `path` option on the GRPC API). Then I had to update the rest of the UI code (which wasn't much anyway) so these [e2e tests](https://github.com/chef/automate/pull/2533/files#diff-f39719fa70b2ecf343fcae6ec13cc4fbR13) would pass. Turned out to not be as independent of a change as I thought.

@msorens I'll take care of the commas after I get buildkite passing, then I'll fix those other files you mentioned [in this branch](https://github.com/chef/automate/compare/bhd/2515/v2-paths-in-tests)

**Comment by msorens** [comment]
🤔 Shouldn't there have been an error about needing to regenerate from proto files due to commit `4ea24d4` ?

**Comment by bcmdarroch** [comment]
I think the compiler considered the comma removal as a cosmetic change since it didn't regenerate any files 🤷‍♀

**Pull request merged**